emote.py

```diff
@@ -14,11 +14,15 @@ def init(self):
 		if self.path is not None:
 			try:
 				if not os.path.isfile(self.path):
-					open(self.path, 'w+')
+					with open(self.path, 'w+') as f:
+						json.dump(self.emotes, f)
 
 				else:
-					self.emotes = json.loads(open(self.path, 'r').read())
+					with open(self.path, 'r') as f:
+						self.emotes = json.loads(f.read())
 
+			except (JSONDecodeError):
```

Comment:
Did you trigger that path? Names python `except` statements are not resolved until an error within the try block is actually raised. It looks to me like you are missing a `json.` in front of `JSONDecodeError`. And although the doc says it exists, that exception does not seem to exist in my python3.4 installation... strange.